### PR TITLE
EVG-6482: run script on all hosts in a distro

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-6482

* Add REST route that allows you to pass a script for all hosts on a distro to execute.
    * Only executes for hosts that are currently running and are not spawned by a user.
* Logs of output/failure appear in host event page.

### CODE REVIEW

File: rest/data/interface.go [hunk 1]
```diff
@@ -135,6 +135,8 @@ type Connector interface {
 	// started by
 	FindHostByIdWithOwner(string, gimlet.User) (*host.Host, error)
 
+	FindHostsByDistroID(string) ([]host.Host, error)
```
Comment (tychoish) [inline rest/data/interface.go]: does it make sense to have these return API hosts rather than host.Hosts

File: units/host_execute.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		j.AddError(err)
+	}
+	output := &util.CappedWriter{
+		Buffer:   &bytes.Buffer{},
+		MaxBytes: 1024 * 1024, // 1MB
+	}
+	if err := j.env.JasperManager().CreateCommand(ctx).Host(hostInfo.Hostname).User(hostInfo.User).
+		ExtendRemoteArgs("-p", hostInfo.Port, "-t", "-t").ExtendRemoteArgs(sshOptions...).
+		SetCombinedWriter(output).
+		AppendArgs("bash", "-s", fmt.Sprintf("<<'EOF'\n%s\nEOF", j.Script)).Run(ctx); err != nil {
```
Comment (tychoish) [inline units/host_execute.go]: I'd not put this in the if statement, just because it's a bit awkward

Comment (tychoish) [inline units/host_execute.go]: as a second commant, should we have a ShellScript() method on command that does bash -s "eof" things?

File: units/host_execute.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+		j.AddError(err)
+		return
+	}
+
+	if j.host.Status != evergreen.HostRunning {
+		grip.Debug(message.Fields{
+			"message": "host is down, not attempting to run script",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+		})
```
Comment (tychoish) [inline units/host_execute.go]: j.ID()

File: units/host_execute.go (hunk 2 above)
Comment (Kimchelly) [reply units/host_execute.go]: I changed this to just use `ShellScript()`.

File: model/host/db.go [hunk 4]
```diff
@@ -467,6 +467,11 @@ func ById(id string) db.Q {
 	return db.Query(bson.D{{Key: IdKey, Value: id}})
 }
 
+func ByDistroIDRunning(distroID string) db.Q {
```
Comment (john-m-liu) [inline model/host/db.go]: Shouldn't this also have a filter to exclude spawn hosts?

File: units/host_execute.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+		j.AddError(err)
+	}
+	output := &util.CappedWriter{
+		Buffer:   &bytes.Buffer{},
+		MaxBytes: 1024 * 1024, // 1MB
+	}
+	err = j.env.JasperManager().CreateCommand(ctx).Host(hostInfo.Hostname).User(hostInfo.User).
+		ExtendRemoteArgs("-p", hostInfo.Port, "-t", "-t").ExtendRemoteArgs(sshOptions...).
+		SetCombinedWriter(output).
+		ShellScript("bash", j.Script).Run(ctx)
```
Comment (john-m-liu) [inline units/host_execute.go]: bash -c ?

File: model/host/db.go (hunk 4 above)
Comment (Kimchelly) [reply model/host/db.go]: I added a filter for the user who started the host.

File: units/host_execute.go (hunk 5 above)
Comment (Kimchelly) [reply units/host_execute.go]: ShellScript is the same as bash -c.

File: model/host/db.go (hunk 4 above)
Comment (john-m-liu) [reply model/host/db.go]: I don't see it. Is it somewhere else?

Comment (Kimchelly) [reply model/host/db.go]: I forgot to push. Sorry!

**Pull request merged**